master/internal/api_tasks.go

```diff
@@ -429,10 +430,33 @@ func constructTaskLogsFilters(req *apiv1.TaskLogsRequest) ([]api.Filter, error)
 		}
 	}
 
-	addInFilter("allocation_id", req.AllocationIds, len(req.AllocationIds))
-	addInFilter("agent_id", req.AgentIds, len(req.AgentIds))
+	addNumFilter := func(field string, values []string) {
+		if values != nil {
+			if slices.Contains(values, "") {
```

Comment:
why are we assuming `values` would only contain one `""` entry? i dont think the api enforces that?

Reply:
I don't think it's necessary to remove the empty string from `FilterOperationInOrNull` so I will remove this concern